client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/QueryOperation.java

```diff
@@ -86,7 +86,7 @@ protected QueryResponse executeOperation(Transport transport) {
             EnumMarshaller encoder = (EnumMarshaller) remoteQuery.getSerializationContext().getMarshaller(value.getClass());
             value = encoder.encode((Enum) value);
          } else if (value instanceof Boolean) {
-            value = (Boolean) value ? 1 : 0;
+            value = value.toString();
```

Comment:
"Booleans values used to be translated to 0/1 literals internally by the DSL builder in remote mode so the underlying Jpql query would differ for the same query in embedded vs remote mode which is not ideal."